core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -5086,6 +5193,13 @@ public BloomFilter getBloomFilter(int size, double falsePositiveRate, long nTwea
                 BloomFilter friendFilter = receivingFromFriendsGroup.getBloomFilter(size, falsePositiveRate, nTweak);
                 filter.merge(friendFilter);
             }
+            for (KeyChainGroupExtension extension : keyChainExtensions.values()) {
+               if (extension.supportsBloomFilters()) {
+                   BloomFilter extensionBloomFilter = extension.getBloomFilter(size, falsePositiveRate, nTweak);
+                   if (extensionBloomFilter != null)
+                       filter.merge(extensionBloomFilter);
+               }
+            }
```

Comment:
This is how the `AuthenticationGroupExtension`, which manages masternode keys, will "inform" the bloom filter about its keys (public keys and public key hashes).